packages/core/src/index.js

```diff
@@ -93,6 +96,13 @@ function replaceComponent(OldType, NewType, resetHookState) {
 				};
 			}
 
+			if (
+				(vnode[VNODE_DOM] && !document.contains(vnode[VNODE_DOM])) ||
+				(!vnode[VNODE_DOM] && !vnode[VNODE_CHILDREN])
```

Comment:
@JoviDeCroock On deep water here, but looking at vnode I find that when it does not have the dom or children directly, it contains a property `__v` that again contains `__k` and `__e` which looks like dom nodes it is checking for. Also a test in preact does grab `getDom` in a similar fashion way https://github.com/preactjs/preact/blob/a75d32adc2332aea154bd6e234293184940abad6/test/browser/components.test.js#L1794 
Would it be possible to avoid a full reload here with a more detailed check? Skipping this it in a project of mine here looks to work out fine for a suspended component.

Reply:
It would be possible but would raise the Preact dependency to require 10.4.0, we could do the checking and throw in case something goe wrong I guess